packages/cel/src/map.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright 2024-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { isReflectMap, type ReflectMap } from "@bufbuild/protobuf/reflect";
+import { CelUint } from "./value/value.js";
+import { celFromMapKey, celFromMapValue, mapKeyFromCel } from "./proto.js";
+
+const privateSymbol = Symbol.for("@bufbuild/cel/map");
+
+type ReadonlyCelMap = ReadonlyMap<bigint | string | boolean | CelUint, unknown>;
+
+/**
+ * A common abstraction for map types.
+ */
+export interface CelMap extends ReadonlyCelMap {
```

Comment:
Can we fold this type into CelMap for simplicity?


```suggestion
/**
 * A common abstraction for map types.
 */
export interface CelMap extends ReadonlyMap<bigint | string | boolean | CelUint, unknown> {
```